packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -759,49 +755,302 @@ class CupertinoActionSheetAction extends StatelessWidget {
   }
 }
 
-class _CupertinoActionSheetCancelButton extends StatefulWidget {
-  const _CupertinoActionSheetCancelButton({
-    this.child,
+// This widget renders the background (pressed or not) of a button, and sends
+// whether the button is pressed to the parent.
+class _ActionSheetButtonBackground extends StatefulWidget {
+  const _ActionSheetButtonBackground({
+    super.key,
+    this.isCancel = false,
+    required this.onStateChange,
+    required this.child,
   });
 
-  final Widget? child;
+  final bool isCancel;
+
+  /// The callback that is called when the button is tapped.
```

Comment:
This comment and variable name don't seem to match

Reply:
I've changed it to:
```dart
  /// Called when the user taps down or lifts up on the button.
  ///
  /// The boolean value is true if the user is tapping down on the button.
```